micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushMeterRegistry.java

```diff
@@ -133,6 +129,14 @@ public void close() {
         stop();
         if (config.enabled() && !isClosed()) {
             publishSafely();
+            try {
+                // block when publishSafely is in progress on another thread
```

Comment:
I refactored with more descriptive method names so that hopefully it is more apparent what is happening from the callsite without so much info in comments. WDYT?